activemodel/lib/active_model/validations/numericality.rb

```diff
@@ -64,14 +62,16 @@ def validate_each(record, attr_name, value)
 
     protected
 
-      def parse_raw_value_as_a_number(raw_value)
-        Kernel.Float(raw_value) if raw_value !~ /\A0[xX]/
+      def is_number?(raw_value)
+        parsed_value = Kernel.Float(raw_value) if raw_value !~ /\A0[xX]/
+        parsed_value.present?
       rescue ArgumentError, TypeError
-        nil
+        false
       end
 
-      def parse_raw_value_as_an_integer(raw_value)
-        raw_value.to_i if raw_value.to_s =~ /\A[+-]?\d+\z/
+      def is_integer?(raw_value)
+        parsed_value = raw_value.to_i if raw_value.to_s =~ /\A[+-]?\d+\z/
```

Comment:
`to_i` will always be `present?` (whether 0 or otherwise) so this can be simplified to 

```ruby
match = raw_value.to_s =~ /\A[+-]?\d+\z/
match.present?
```